app/ai_backend/test_genesis_api.py

```
@@ -0,0 +1,767 @@
+            return mock_response
+        
+        with patch('aiohttp.ClientSession.post', side_effect=mock_post_with_recovery):
+            with patch('asyncio.sleep'):  # Speed up test
+                async with GenesisAPIClient(**config) as client:
+                    try:
+                        result = await client.create_chat_completion(
+                            messages=[ChatMessage(role="user", content="Test")],
+                            model_config=ModelConfig(name="test-model")
+                        )
+                        # Should not reach here due to rate limit
+                        assert False, "Expected RateLimitError"
```

Ruff (0.11.9)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)